worldedit-core/src/main/java/com/fastasyncworldedit/core/command/tool/FeaturePlacer.java

```diff
@@ -0,0 +1,77 @@
+package com.fastasyncworldedit.core.command.tool;
+
+import com.fastasyncworldedit.core.configuration.Caption;
+import com.sk89q.worldedit.EditSession;
+import com.sk89q.worldedit.LocalConfiguration;
+import com.sk89q.worldedit.LocalSession;
+import com.sk89q.worldedit.MaxChangedBlocksException;
+import com.sk89q.worldedit.command.tool.BlockTool;
+import com.sk89q.worldedit.entity.Player;
+import com.sk89q.worldedit.extension.platform.Actor;
+import com.sk89q.worldedit.extension.platform.Platform;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.util.Direction;
+import com.sk89q.worldedit.util.Location;
+import com.sk89q.worldedit.world.generation.ConfiguredFeatureType;
+
+import javax.annotation.Nullable;
+
+/**
+ * Places a feature
+ *
+ * @since TODO
+ */
+public class FeaturePlacer implements BlockTool {
+
+    private final ConfiguredFeatureType feature;
+
+    /**
+     * New instance
+     *
+     * @since TODO
+     */
+    public FeaturePlacer(ConfiguredFeatureType feature) {
+        this.feature = feature;
+    }
+
+    @Override
+    public boolean canUse(Actor player) {
+        return player.hasPermission("worldedit.tool.feature");
+    }
+
+    @Override
+    public boolean actPrimary(
+            Platform server,
+            LocalConfiguration config,
+            Player player,
+            LocalSession session,
+            Location clicked,
+            @Nullable Direction face
+    ) {
+
+        try (EditSession editSession = session.createEditSession(player)) {
+            try {
+                boolean successful = false;
+
+                final BlockVector3 pos = clicked.toVector().add().toBlockPoint();
+                for (int i = 0; i < 10; i++) {
```

Comment:
What's the repetition for?

Reply:
I believe it exists/existed somewhere in Minecaaft or upstream? There is often a random aspect to a feature or structure (e.g. tree height) that means it might fail to be constructed occasionally. This is an arbitrary way to try a few times basically